app/helpers/checkouts_helper.rb

```diff
@@ -0,0 +1,60 @@
+module CheckoutsHelper
+  def active_step(current_step)
+    'active' if current_step == step
+  end
+
+  def address_error?(field:, tag:)
+    return @billing.errors.include?(field) if @billing && tag == 'billing'
+    @shipping.errors.include?(field) if @shipping && tag == 'shipping'
```

Comment:
Move `@shipping && tag == 'shipping'` to private method. You use this condition in several methods.